### ISSUE TITLE
Crash when adding a module selection that contains a I-Dec to the palette.

### ISSUE THREAD

**Issue by Pouf842** [PleasingFungus/Silicon-Zeroes#36]
It does either if I add this by the new Right-click UI, or by the palette menu.
It does not crash if I don't include the I-Dec in the selection.

No crash report, only "Processus stopped working".

The selection included a counter (Latch, Adder and Number 1) and a I-Dec, all connected.

Error.txt :

```
1376044: CreateTexture(D3DPOOL_DEFAULT): INVALIDCALL
58551401: CreateTexture(D3DPOOL_DEFAULT): INVALIDCALL
525: CreateIconIndirect(): Paramètre incorrect.


178: CreateIconIndirect(): Paramètre incorrect.


175: CreateIconIndirect(): Paramètre incorrect.


186: CreateIconIndirect(): Paramètre incorrect.


169: CreateIconIndirect(): Paramètre incorrect.
```

but I suspect this to be from previous crashes (Cursor crash at start), so it seems there's nothing here for this issue.

Edit : error.txt confirmed to not change after other same crashes.

**Comment by Pouf842** [comment]
The + button in the palette menu behave the exact same way.

After further tests, it seems to be the "I-Dec" module that provokes the crash, since if the selection does not include it, the game does not crash.

**Comment by PleasingFungus** [comment]
Issue with the name parsing code.

**Comment by PleasingFungus** [comment]
Fixed in afd9c877b9, thanks!
